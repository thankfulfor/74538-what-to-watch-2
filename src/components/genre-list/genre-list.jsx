import * as React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {FilmList} from '../film-list/film-list-1.jsx';
import GenreTab from '../genre-tab/genre-tab.jsx';
import {ShowMoreButton} from '../show-more-button/show-more-button.jsx';

import {increaseCountFilmsShowAction} from '../../actions/increase-count-films-show.js';

import {getFilteredFilms, getGenres} from '../../selector/selectors.js';

export const GenreList = (props) => {
  const {onShowMoreButtonClick, filteredFilms, genres} = props;

  const showMoreButtonClickHandler = (evt) => {
    evt.preventDefault();
    onShowMoreButtonClick();
  };

  return (
    <section className="catalog">
      <h2 className="catalog__title visually-hidden">Catalog</h2>

      <GenreTab genres={genres} />

      <FilmList films={filteredFilms} />

      <ShowMoreButton onClick={showMoreButtonClickHandler} />
    </section>
  );
};

GenreList.propTypes = {
  films: PropTypes.array.isRequired,
  filteredFilms: PropTypes.array.isRequired,
  genre: PropTypes.string.isRequired,
  countFilmsShow: PropTypes.number.isRequired,
  onShowMoreButtonClick: PropTypes.func.isRequired,
  genres: PropTypes.array.isRequired
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  films: state.films,
  genre: state.genre,
  countFilmsShow: state.countFilmsShow,
  filteredFilms: getFilteredFilms(state),
  genres: getGenres(state)
});

const mapDispatchToProps = (dispatch) => ({
  onShowMoreButtonClick: () => {
    dispatch(increaseCountFilmsShowAction());
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(GenreList);

