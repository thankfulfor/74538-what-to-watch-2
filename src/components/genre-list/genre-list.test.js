import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import {GenreList} from './genre-list.jsx';

const films = [{
  previewImage: `img/we-need-to-talk-about-kevin.jpg`,
  name: `We need to talk about Kevin`,
  genre: `drama`,
  previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/b/b3/Big_Buck_Bunny_Trailer_400p.ogv`
}];

const genre = `All genres`;

Enzyme.configure({adapter: new Adapter()});

it(`Is genre catalog rendered`, () => {
  const tree = shallow(
      <GenreList
        films={films}
        genre={genre}
      />);

  expect(toJson(tree)).toMatchSnapshot();
});
