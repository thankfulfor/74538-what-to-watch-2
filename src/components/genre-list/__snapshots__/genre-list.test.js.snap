// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is genre catalog rendered 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <Connect(GenreTab)
    genres={
      Array [
        "All genres",
      ]
    }
  />
  <FilmList
    films={
      Array [
        Object {
          "genre": "drama",
          "name": "We need to talk about Kevin",
          "previewImage": "img/we-need-to-talk-about-kevin.jpg",
          "previewVideoLink": "https://upload.wikimedia.org/wikipedia/commons/b/b3/Big_Buck_Bunny_Trailer_400p.ogv",
        },
      ]
    }
  />
  <ShowMoreButton
    onClick={[Function]}
  />
</section>
`;
